### ISSUE TITLE
DirectionalSplitting not twitstet blockDim

### ISSUE THREAD

**Issue by psychocoderHPC** [ComputationalRadiationPhysics/picongpu#889]
The current implementation of `Directional Splitting` twist the axes and run three kernel to calculate new field. The used algorithm `ForEachBlock` had one compile time vector with the Blocksize (equal to SuperCellSize).
https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/src/picongpu/include/fields/MaxwellSolver/DirSplitting/DirSplitting.hpp#L78

My question is: Is this right that the `BlockDim` is not twistet? For me it looks like we get memory access in the GUARD were we only calculate the CORE+BORDER domain. Might be also out of memory access in some cases.

IMO and my understanding of the implementation we must twist the `BlockDim` in the same way we does it with the `gridSize`. 
The reason is that our default SuperCell is 8x8x4 cells. We twist the fields thus every direction (x/y/z) is in one run at in kernel direction x. The kernel traverse over x and the shared memory size is create out of `BlockDim` by extent the x direction with two cells. In the default SuperCell case 10x8x4. If in one of the three runs field direction `z` is mapped to `x` and the simulation size is `24x24x12` `12` is mapped to `x` and traversed with a block 8x8x4 threads.

@Heikman this might be very important   for @ax3l

**Comment by psychocoderHPC** [comment]
I change this to a bug `cuda-memcheck` confirmed that we get out of memory access.

Directional Splitting without particles and laser plane wave:
```
cuda-memcheck --report-api-errors no  ./picongpu -d 1 1 1 -g 24 64 12 -s 1 --periodic 1 0 1
```
```BASH
========= CUDA-MEMCHECK
========= Invalid __global__ read of size 4
=========     at 0x000006f0 in _ZN5PMacc9algorithm6kernel6detail18kernelForeachBlockINS2_13Sph
ericMapperILi3ENS_4math2CT6VectorIN4mpl_10integral_cIiLi8EEESA_NS9_IiLi4EEEEENS8_5void_EEENS_6
cursor6CursorINSF_14MarkerAccessorINSG_INSF_17TwistAxesAccessorINSF_12BufferCursorINS5_6Vector
IfLi3ENS5_16StandartAccessorENS5_17StandartNavigatorENS5_6detail17Vector_componentsEEELi3EEENS
6_3IntILi2ELi0ELi1EEEEENSF_2CT18TwistAxesNavigatorISS_Li3EEESQ_EEEENSF_15CursorNavigatorESX_EE
S10_N8picongpu18DirSplittingKernelISC_EEEEvT_T0_T1_T2_
=========     by thread (7,3,2) in block (0,0,13)
=========     Address 0x13045afabc is out of bounds
=========     Saved host backtrace up to driver entry point at kernel launch time
=========     Host Frame:/opt/pkg/devel/cuda/6.5/lib64/libcuda.so.1 (cuLaunchKernel + 0x2cd) [
0x150cad]
=========     Host Frame:/opt/pkg/devel/cuda/6.5/lib64/libcudart.so.6.5 [0xf5d8]
=========     Host Frame:/opt/pkg/devel/cuda/6.5/lib64/libcudart.so.6.5 (cudaLaunch + 0x143) [
0x36833]
=========     Host Frame:./picongpu (_ZN5PMacc9algorithm6kernel12ForeachBlockINS_4math2CT6Vect
orIN4mpl_10integral_cIiLi8EEES8_NS7_IiLi4EEEEESA_EclINS_4zone11SphericZoneILi3EEENS_6cursor6Cu
rsorINSG_14MarkerAccessorINSH_INSG_17TwistAxesAccessorINSG_12BufferCursorINS3_6VectorIfLi3ENS3
_16StandartAccessorENS3_17StandartNavigatorENS3_6detail17Vector_componentsEEELi3EEENS4_3IntILi
2ELi0ELi1EEEEENSG_2CT18TwistAxesNavigatorIST_Li3EEESR_EEEENSG_15CursorNavigatorESY_EES11_N8pic
ongpu18DirSplittingKernelISA_EEEEvRKT_T0_T1_RKT2_ + 0x2cc) [0xc47ec]
=========     Host Frame:./picongpu (_ZNK8picongpu12dirSplitting12DirSplitting20update_beforeC
urrentEj + 0xc09) [0xcb029]
=========     Host Frame:./picongpu (_ZN8picongpu12MySimulation10runOneStepEj + 0xff) [0xcb73f
]
=========     Host Frame:./picongpu (_ZN5PMacc16SimulationHelperILj3EE15startSimulationEv + 0x
1c0) [0xb0f90]
=========     Host Frame:./picongpu (_ZN8picongpu17SimulationStarterINS_21InitialiserControlle
rENS_16PluginControllerENS_12MySimulationEE5startEv + 0x90) [0xe3950]
=========     Host Frame:./picongpu (main + 0x62) [0x777e2]
=========     Host Frame:/lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main + 0xf5) [0x21ec5]
=========     Host Frame:./picongpu [0x77baf]
=========
========= Invalid __global__ read of size 4
```

**Comment by psychocoderHPC** [comment]
Add fix in #890

**Comment by ax3l** [comment]
merci!
